instagram/src/components/addPost.js

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import '../index.css'
+
+class AddPost extends React.Component {
+
+  // constructor(){
+  //   super();
+    // super like polymorphism - App as the master class, which this class refers to.
+
+    state = {
+      caption: '',
+      image: '',
+      posts: []
+      // make a state for the incoming data when API is called.
+    };
+
+    showInstaPost = () => {
+      this.getInstaPost();
+    };
+
+    getInstaPost = () => {
+      axios.get('http://localhost:8080/api/')
+      .then((response) => {
+        const data = response.data;
+        this.setState({ posts: data });
+        console.log('Data has been received!');
+      })
+       .catch(() => {
+         alert('Error retrieving data!');
+     });
+    }
+
+    handleFileSelect = (e) => {
+          var files = e.target.files;
+          var f = files[0];
+          var reader = new FileReader();
+
+            reader.onload = (function(theFile) {
+                  return function(e) {
+                    document.getElementById('list').innerHTML = ['<img src="', e.target.result,'" title="', theFile.name, '" width="50" />'].join('');
+                  };
+            })(f);
+            reader.readAsDataURL(f);
+            this.onChange();
+      }
+
+
+    handleChange = ({ target }) => {
+      const { name, value } = target;
+      this.setState({ [name]: value });
+    };
+
+
+   submit = (event) => {
+    event.preventDefault();
+// prevents the browser from refreshing
+
+    const serverData = {
+      caption: this.state.caption,
+      image: this.state.image
+    };
+
+    axios({
```

Comment:
And also good to see the use of axios here, i think we need to include this into node4code as it's not talking to the backend as it is, axios will help that so its good to see it here!